### DISCUSSION TITLE
Feature Request: connect to duckdb via http api

### DISCUSSION THREAD

**Discussion by fyears** [duckdb/duckdb#12561] [Ideas]
It will be great if duckdb can be connected via http api. For example, it’s good for edge computing.

(attentions: i know duckdb can connect **to** http or s3 data. but I am proposing connecting to duckdb here. )

Notably, libsql, a fork of sqlite, provides this function: https://github.com/tursodatabase/libsql/tree/main/libsql-server

And webdis (https://github.com/nicolasff/webdis) adds the interface to redis.

**Comment by lmangani** [comment]
There are quite a few projects adding light HTTP APIs on top of DuckDB around on github. 
One example is our project [quackpipe](https://github.com/metrico/quackpipe) which emulates the Clickhouse HTTP API and query interface but there any many. 
Hope this helps!

**Reply by mossabotmanbasir** [reply]
Saved the day! Thanks for this guys.

**Comment by udota** [comment]
bro, honestly i find the best sync plugin, and i want to ask u a question, can i put Tor briges on this plugin? because i want secure connection with other clouds
